src/config.c

```diff
@@ -233,14 +222,20 @@ typedef union typeData {
     numericConfigData numeric;
 } typeData;
 
+typedef int (*apply_fn)(const char **err);
 typedef struct typeInterface {
     /* Called on server start, to init the server with default value */
     void (*init)(typeData data);
-    /* Called on server startup and CONFIG SET, returns 1 on success, 0 on error
-     * and can set a verbose err string, update is true when called from CONFIG SET */
-    int (*set)(typeData data, sds *argv, int argc, int update, const char **err);
-    /* Called on CONFIG GET, required to add output to the client */
-    void (*get)(client *c, typeData data);
+    /* Called on server startup and CONFIG SET, returns 1 on success,
+     * 2 meaning no actual change done, 0 on error and can set a verbose err
+     * string */
+    int (*set)(typeData data, sds *argv, int argc, const char **err);
+    /* Optional: called after `set()` to apply the config change. Used only in
+     * the context of CONFIG SET. Returns 1 on success, 0 on failure.
+     * Optionally set err to a static error string. */
+    apply_fn apply;
```

Comment:
nit: use explicit function type to be consistent with the rest of the fields.

Reply:
It's just that I use this type for an array of (unique) function pointers to be applied after _set_ or during restore after failure. I wanted to avoid the risk of someone changing the function type in one place and not the other.